### PR TITLE
Refactor profile info

### PR DESCRIPTION
**Pull request by snim2**
This PR refactors the tool visualisers so that tool information is stored on nodes rather than in _ad hoc_ dictionaries.

### CODE REVIEW

File: lib/eco/export/simple_language.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        # Run this command:
+        #     $ cd $GRAAL_WORKSPACE/graal-compiler
+        #     $ ../../mx/mx --vm graal sl tempfile.sl
+        return subprocess.Popen(["../../mx/mx", "--vm", "graal", "sl", f[1]],
+                                cwd=working_dir,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.STDOUT,
+                                bufsize=0)
+
+    def _profile(self):
```
Comment (ltratt) [inline lib/eco/export/simple_language.py:62]: Double indented body?

File: lib/eco/treemanager.py [hunk 2]
```diff
@@ -289,9 +290,9 @@ def __init__(self):
         self.savenextparse = False
         self.saved_lines = {}
         self.saved_parsers = {}
-        self.profile_data = {} # line no -> (float, node) (raw data)
-        self.profile_map = {}  # node -> str
-        self.profile_is_dirty = False
+
+        self.tool_information_is_dirty = False
```
Comment (ltratt) [inline lib/eco/treemanager.py]: Can we maybe use a shorter name than `tool_information_is_dirty`? Maybe `tool_data_is_dirty` or something like that?

**Comment by ltratt** [comment]
A couple of minor comments (plus let's make the whitespace consistent with the rest of Eco), but looks good overall.

**Comment by snim2** [comment]
Ready to squash?

**Comment by ltratt** [comment]
Yep, let's squash!

**Pull request merged**